geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaProducerFeatureStore.scala

```diff
@@ -64,86 +64,83 @@ class KafkaProducerFeatureStore(entry: ContentEntry,
     case _              => super.removeFeatures(filter)
   }
 
-  def clearFeatures(): Unit = {
-    val msg = GeoMessage.clear()
-    logger.debug("sending message: {}", msg)
+  def clearFeatures(): Unit = writerPool.withResource { fw => fw.send(GeoMessage.clear()) }
 
-    val encoder = new KafkaGeoMessageEncoder(sft)
-    producer.send(encoder.encodeClearMessage(topic, msg))
+  override def getWriterInternal(query: Query, flags: Int) = {
```

Comment:
it's part of the content data store API - I believe it indicates whether this is a modify or append operation. But since we only support modify by ID, we ignore it. Technically if the query was 'INCLUDE' and the flag was modify, we should return all the existing features for modification. Since that's not possible due to the producer not having any features cached, we treat it the same as append.